app/src/main/java/com/expidev/gcmapp/MainActivity.java

```diff
@@ -272,6 +296,95 @@ private void login()
             loginDialogFragment.show(fm.beginTransaction().addToBackStack("loginDialog"), "loginDialog");
         }
     }
+    
+    private void getCurrentAssignment()
+    {
+        if (currentAssignment == null)
+        {
+            MinistriesDao ministriesDao = MinistriesDao.getInstance(this);
+            
+            if (associatedMinistries == null)
+            {
+                chosenMinistry = preferences.getString("chosen_ministry", null);
+                
+                associatedMinistries = ministriesDao.retrieveAssociatedMinistriesList();
+            }
+
+            if (associatedMinistries == null || associatedMinistries.size() == 0)
+            {
+                Log.w(TAG, "No Associated Ministries");
+                return;
+            }
+
+            if (chosenMinistry == null)
+            {
+                SharedPreferences.Editor editor = preferences.edit();
+                
+                int i = 0;
+                boolean parent = false;
+                do
+                {
+                    if (associatedMinistries.get(i).getParentId() == null) parent = true;
+                }while (!parent);
+                    
+                currentMinistry = associatedMinistries.get(i);
+                currentMinistry = associatedMinistries.get(1);
```

Comment:
Yes. I am currently working on a better version of this method that runs on a background thread.